browser/brave_content_browser_client.cc

```diff
@@ -454,18 +453,28 @@ BraveContentBrowserClient::CreateURLLoaderThrottles(
       request, browser_context, wc_getter, navigation_ui_data,
       frame_tree_node_id);
 #if BUILDFLAG(ENABLE_SPEEDREADER)
+  using DistillState = speedreader::SpeedreaderTabHelper::DistillState;
   content::WebContents* contents = wc_getter.Run();
   if (!contents) {
     return result;
   }
   auto* tab_helper =
       speedreader::SpeedreaderTabHelper::FromWebContents(contents);
-  if (tab_helper && tab_helper->IsActiveForMainFrame() &&
+  if (!tab_helper)
+    return result;
```

Comment:
quick return like this looks error-prone: imagine someone adds another throttle below the speedreader one. So I'd suggest we stick to "positive" checks

Reply:
fixed